content/manuals/build/cache/garbage-collection.md

```diff
@@ -8,89 +8,266 @@ aliases:
+## Garbage collection policies
+
+GC policies define a set of rules that determine how the build cache is managed
+and cleaned up. These policies include criteria for when to remove cache
+entries, such as the age of the cache, the amount of space being used, and
+other constraints like ensuring a minimum amount of free disk space or
+reserving a specific percentage of the disk for build cache.
+
+Each GC policy is evaluated in sequence, starting with the most specific
+criteria, and proceeds to broader rules if previous policies do not resolve the
+issue. This lets BuildKit prioritize cache entries, preserving the most
+valuable cache while ensuring the system maintains performance and
+availability.
+
+The default GC policies are (approximately):
+
+1. Remove cache that can be easily regenerated, such as build contexts from
+   local directories or remote Git repositories, and cache mounts, if hasn't
+   been used for more than 48 hours.
+2. Remove cache that hasn't been used in a build for more than 60 days.
+3. Remove unshared cache that exceeds the build cache size limit. Unshared
+   cache records refers to layer blobs that are not used by other resources
+   (typically, as image layers).
+4. Remove any build cache that exceeds the build cache size limit.
+
+The precise algorithm and the means of configuring the policies differ slightly
+depending on what kind of builder you're using. Refer to
+[Configuration](#configuration) for more details.
 
 ## Configuration
 
-Depending on the [driver](../builders/drivers/_index.md) used by your builder instance,
-the garbage collection will use a different configuration file.
+> [!NOTE]
+> If you're satisfied with the default garbage collection behavior and don't
+> need to fine-tune its settings, you can skip this section. Default
+> configurations work well for most use cases and require no additional setup.
+
+Depending on the type of [build driver](../builders/drivers/_index.md) you use,
+you will use different configuration files to change the builder's GC settings:
+
+- If you use the default builder for Docker Engine (the `docker` driver), use
+  the [Docker daemon configuration file](#docker-daemon-configuration-file).
+- If you use a custom builder, use a [BuildKit configuration file](#buildkit-configuration-file).
+
+### Docker daemon configuration file
+
+If you're using the default [`docker` driver](../builders/drivers/docker.md),
+GC is configured in the [`daemon.json` configuration file](/reference/cli/dockerd.md#daemon-configuration-file),
+or if you use Docker Desktop, in [**Settings > Docker Engine**](/manuals/desktop/settings-and-maintenance/settings.md).
+
+The following snippet shows the default builder configuration for the `docker`
+driver for Docker Desktop users:
+
+```json
+{
+  "builder": {
+    "gc": {
+      "defaultKeepStorage": "20GB",
+      "enabled": true
+    }
+  }
+}
+```
+
+The `defaultKeepStorage` option configures the size limit of the build cache,
```

Comment:
From BuildKit v0.17 and https://github.com/moby/moby/pull/48720 these defaults are different with reserved,free,max all being configured by default.